tests/providers/amazon/aws/hooks/test_s3.py

```diff
@@ -85,22 +85,42 @@ def test_parse_s3_url(self):
         parsed = S3Hook.parse_s3_url("s3://test/this/is/not/a-real-key.txt")
         assert parsed == ("test", "this/is/not/a-real-key.txt"), "Incorrect parsing of the s3 url"
 
+    def test_parse_s3_url_with_space_and_octothorpe(self):
```

Comment:
nit: all these tests could be merged into one using `@pytest.mark.parametrize`

Reply:
I agree with this and would like to see that done if you are up for it @GlenboLake, but I won't block for it either.

Reply:
Done. If the formatting is to your liking, I'll squash it. If you'd prefer a pair of `parametrize` marks I could do that too.

Reply:
Thanks!